pkg/apis/resource/validation/validation.go

```diff
@@ -433,179 +512,244 @@ func validateNodeName(name string, fldPath *field.Path) field.ErrorList {
-	allErrs := corevalidation.ValidateObjectMetaUpdate(&resourceClaimParameters.ObjectMeta, &oldResourceClaimParameters.ObjectMeta, field.NewPath("metadata"))
-	requestStored := apiequality.Semantic.DeepEqual(oldResourceClaimParameters.DriverRequests, resourceClaimParameters.DriverRequests)
-	allErrs = append(allErrs, validateResourceClaimParameters(resourceClaimParameters, requestStored)...)
-	return allErrs
+func validateQuantity(quantity apiresource.Quantity, fldPath *field.Path) field.ErrorList {
+	// Any parsed quantity is valid.
+	return nil
 }
 
-// ValidateResourceClassParameters tests if a ResourceClassParameters object is valid for creation.
-func ValidateResourceClassParameters(parameters *resource.ResourceClassParameters) field.ErrorList {
-	return validateResourceClassParameters(parameters, false)
+func validateQualifiedName(name resource.QualifiedName, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if name == "" {
+		allErrs = append(allErrs, field.Required(fldPath, "name required"))
+		return allErrs
+	}
+
+	// Naming the two parts in a field path is tricky. Treating them as a child field
+	// is not quite right, but close enough...
+	parts := strings.Split(string(name), "/")
+	switch len(parts) {
+	case 1:
+		allErrs = append(allErrs, validateCIdentifier(parts[0], fldPath.Child("identifier"))...)
+	case 2:
+		allErrs = append(allErrs, validateDriverName(parts[0], fldPath.Child("domain"))...)
+		allErrs = append(allErrs, validateCIdentifier(parts[1], fldPath.Child("identifier"))...)
+
+	}
+	return allErrs
 }
 
-func validateResourceClassParameters(parameters *resource.ResourceClassParameters, filtersStored bool) field.ErrorList {
-	allErrs := corevalidation.ValidateObjectMeta(&parameters.ObjectMeta, true, apimachineryvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
-	allErrs = append(allErrs, validateClassParameters(parameters.GeneratedFrom, field.NewPath("generatedFrom"))...)
-	allErrs = append(allErrs, validateResourceFilters(parameters.Filters, field.NewPath("filters"), filtersStored)...)
-	allErrs = append(allErrs, validateVendorParameters(parameters.VendorParameters, field.NewPath("vendorParameters"))...)
+func validateFullyQualifiedName(name resource.FullyQualifiedName, fldPath *field.Path) field.ErrorList {
+	allErrs := validateQualifiedName(resource.QualifiedName(name), fldPath)
+	if !strings.Contains(string(name), "/") {
+		allErrs = append(allErrs, field.Required(fldPath.Child("domain"), "a fully-qualified name must include a domain prefix"))
+	}
 	return allErrs
 }
 
-func validateResourceFilters(filters []resource.ResourceFilter, fldPath *field.Path, filtersStored bool) field.ErrorList {
+// cIdentifierFmt is the same as in https://github.com/google/cel-spec/blob/master/doc/langdef.md#overview for IDENT.
+const cIdentifierFmt string = "[_a-zA-Z][_a-zA-Z0-9]*"
```

Comment:
We have `IsCIdentifier` which perhaps nobody at all calls?

Reply:
Funny, that almost looks like the code that I wrote :grin: 

Let's use the existing one.